cachi2/core/package_managers/yarn/project.py

```diff
@@ -49,220 +45,33 @@ def __init__(self, path: RootedPath, data: dict[str, Any]) -> None:
-    @property
-    def pnp_data_path(self) -> Optional[str]:
-        """Get the pnpDataPath configuration."""
-        return self._data.get("pnpDataPath", None)
-
-    @pnp_data_path.setter
-    def pnp_data_path(self, path: Optional[str]) -> None:
-        self._data["pnpDataPath"] = path
-
-    @property
-    def pnp_mode(self) -> Optional[PnpMode]:
-        """Get the pnpMode configuration."""
-        return self._data.get("pnpMode", None)
-
-    @pnp_mode.setter
-    def pnp_mode(self, mode: Optional[PnpMode]) -> None:
-        self._data["pnpMode"] = mode
-
-    @property
-    def pnp_unplugged_folder(self) -> Optional[str]:
-        """Get the PnP unplugged folder."""
-        return self._data.get("pnpUnpluggedFolder", None)
-
-    @pnp_unplugged_folder.setter
-    def pnp_unplugged_folder(self, folder: Optional[str]) -> None:
-        self._data["pnpUnpluggedFolder"] = folder
-
-    @property
-    def ignore_path(self) -> Optional[bool]:
-        """Get the ignorePath configuration."""
-        return self._data.get("ignorePath", None)
-
-    @ignore_path.setter
-    def ignore_path(self, ignore_path: Optional[bool]) -> None:
-        self._data["ignorePath"] = ignore_path
-
-    @property
-    def unsafe_http_whitelist(self) -> list[str]:
-        """Get the whitelisted urls that can be accessed via http.
-
-        Returns an empty array in case there are none defined.
-        """
-        return self._data.get("unsafeHttpWhitelist", [])
-
-    @unsafe_http_whitelist.setter
-    def unsafe_http_whitelist(self, urls: list[str]) -> None:
-        self._data["unsafeHttpWhitelist"] = urls
-
-    @property
-    def node_linker(self) -> NodeLinker:
-        """Get the nodeLinker configuration."""
-        return self._data.get("nodeLinker", None)
-
-    @node_linker.setter
-    def node_linker(self, node_linker: Optional[NodeLinker]) -> None:
-        self._data["nodeLinker"] = node_linker
-
-    @property
-    def plugins(self) -> list[Plugin]:
-        """Get the configured plugins.
-
-        Returns an empty array in case there are none defined.
-        """
-        return self._data.get("plugins", [])
-
-    @plugins.setter
-    def plugins(self, plugins: list[Plugin]) -> None:
-        self._data["plugins"] = plugins
-
-    @property
-    def virtual_folder(self) -> Optional[str]:
-        """Get the virtualFolder configuration."""
-        return self._data.get("virtualFolder", None)
-
-    @virtual_folder.setter
-    def virtual_folder(self, folder: Optional[str]) -> None:
-        self._data["virtualFolder"] = folder
-
-    @property
-    def registry_server(self) -> str:
-        """Get the globally configured registry server.
+    def __getitem__(self, key: str) -> Any:
+        try:
+            return self.data[key]
+        except KeyError:
+            try:
+                # try the defaults potentially raising KeyError
+                return self._defaults[key]
+            except KeyError:
+                return None
```

Comment:
I wonder if adding something along these lines would be a good idea:
```python
def __getattr__(self, attr):
    if attr in self.__dict__:
         return self.__dict__[attr]
    return self[attr]

def __setattr__(self, attr, value):
    self.data[attr] = value
```
This would let keeping most client code intact.

Reply:
This is a terrible idea, at least the way I posted it, it takes more than the snippet to make it work properly.